website/source/docs/guides/monitoring-telegraf.html.md

```diff
@@ -0,0 +1,315 @@
+  delete_sets = true
+  delete_timings = true
+  percentiles = [90]
+  metric_separator = "_"
+  parse_data_dog_tags = true
+  allowed_pending_messages = 10000
+  percentile_limit = 1000
+```
+
+The full reference to all the available statsd-related options in Telegraf is [here][telegraf-statsd-input].
+
+Now we can configure inputs for things like CPU, memory, network I/O, and disk I/O. Most of them don't require any configuration, but make sure the `interfaces` list in `inputs.net` matches the interface names you see in `ifconfig`.
+
+```ini
+[[inputs.cpu]]
+  percpu = true
+  totalcpu = true
+  collect_cpu_time = false
+
+[[inputs.disk]]
+  # mount_points = ["/"]
+  # ignore_fs = ["tmpfs", "devtmpfs"]
+
+[[inputs.diskio]]
+  # devices = ["sda", "sdb"]
+  # skip_serial_number = false
+
+[[inputs.kernel]]
+  # no configuration
+
+[[inputs.linux_sysctl_fs]]
+  # no configuration
+
+[[inputs.mem]]
+  # no configuration
+
+[[inputs.net]]
+  interfaces = ["enp0s*"]
+
+[[inputs.netstat]]
+  # no configuration
+
+[[inputs.processes]]
+  # no configuration
+
+[[inputs.swap]]
+  # no configuration
+
+[[inputs.system]]
+  # no configuration
+```
+
+Another useful plugin is the [procstat][telegraf-procstat-input] plugin, which reports metrics for processes you select:
+
+```ini
+[[inputs.procstat]]
+  pattern = "(consul)"
+```
+
+Telegraf even includes a [plugin][telegraf-consul-input] that monitors the health checks associated with the Consul agent, using Consul API to query the data.
+
+It's importatnt to note: the plugin itself will not report the telemetry, Consul will report those stats already using StatsD protocol.
+
+```ini
+[[inputs.consul]]
+  address = "localhost:8500"
+  scheme = "http"
+```
+
+[telegraf-install]: https://docs.influxdata.com/telegraf/v1.6/introduction/installation/
+[telegraf-consul-input]: https://github.com/influxdata/telegraf/tree/release-1.6/plugins/inputs/consul
+[telegraf-statsd-input]: https://github.com/influxdata/telegraf/tree/release-1.6/plugins/inputs/statsd
+[telegraf-procstat-input]: https://github.com/influxdata/telegraf/tree/release-1.6/plugins/inputs/procstat
+[telegraf-input-plugins]: https://docs.influxdata.com/telegraf/v1.6/plugins/inputs/
+
+## Telegraf Configuration for Consul
+
+Asking Consul to send telemetry to Telegraf is as simple as adding a `telemetry` section to your agent configuration:
+
+```json
+{
+  "telemetry": {
+    "dogstatsd_addr": "localhost:8125",
+    "disable_hostname": true
+  }
+}
+```
+
+As you can see, we only need to specify two options. The `dogstatsd_addr` specifies the hostname and port of the
+statsd daemon.
+
+Note that we specify DogStatsD format instead of plain statsd, which tells Consul to send [tags][tagging]
+with each metric. Tags can be used by Grafana to filter data on your dashboards (for example, displaying only
+the data for which `role=consul-server`. Telegraf is compatible with the DogStatsD format and allows us to add
+our own tags too, as you'll see later.
```

Comment:
Is it worth mentioning that whether you want tags or not probably depends on your actual metrics backend? If you are sending all this to graphite for example ~you need to stop reading an reconsider~  it won't work. Might be a complication people can work out for themselves though?